### PR TITLE
Add LFX footer to all dashboards

### PR DESCRIPTION
**Pull request by Sameh16**
# Changes proposed ✍️

### What
Add LFX footer to all dashboards

### How
copilot:walkthrough

## Checklist ✅
- [ ] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screenshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [ ] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

## Release Notes

- **New Features**
	- Introduced a new footer component that manages visibility based on page lifecycle.
	- Integrated a new UI library (`@linuxfoundation/lfx-ui-core`) for enhanced design elements.

- **Layout Enhancements**
	- Updated layout structure for better content alignment and footer positioning.
	- Modified main container styling to improve responsiveness.

- **State Management**
	- Added a new store for managing footer visibility, enhancing user experience.

### CODE REVIEW

**Comment by Sameh16** [comment]
@coderabbitai review

**Comment by Sameh16** [comment]
@coderabbitai full review

**Comment by Sameh16** [comment]
@coderabbitai full review

File: frontend/src/modules/dashboard/pages/dashboard-page.vue [hunk 1]
```diff
@@ -35,6 +35,9 @@
           </div>
         </div>
       </div>
+      <div class="mt-auto mb-6">
+        <lfx-footer class="px-2 max-w-3xl" />
+      </div>
```
Comment (joanagmaia) [inline frontend/src/modules/dashboard/pages/dashboard-page.vue:40]: Is it supposed not to be centered here?
Also, dashboard also uses layout I believe. Couldn't we re-use the same footer in all pages?
Screenshot 2024-12-04 at 08 59 03

Comment (Sameh16) [reply frontend/src/modules/dashboard/pages/dashboard-page.vue]: yeah, that's why we have 2 footer one in layout and one in dashboard

Comment (Sameh16) [reply frontend/src/modules/dashboard/pages/dashboard-page.vue]: fixed

File: frontend/src/modules/layout/footer-store.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+export const useFooterStore = defineStore('footer-store', {
+  state: () => ({
+    visible: true,
+  }),
+  actions: {
+    setVisibility(visible: boolean) {
+      this.visible = visible;
+    },
+  },
+});
```
[the remark this answers is not in the document]
Comment (Sameh16) [reply frontend/src/modules/layout/footer-store.ts]: Done

**Review by joanagmaia — APPROVED** [review]

**Pull request merged**